File: flower/src/main/java/com/kharitonov/flower/controllers/UserController.java
package com.kharitonov.flower.controllers;

import com.kharitonov.flower.contracts.ApiRouter;
import com.kharitonov.flower.dtos.UserDTO;
import com.kharitonov.flower.entities.UserEntity;
import com.kharitonov.flower.security.Role;
import com.kharitonov.flower.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Optional;

@RestController
@RequestMapping(ApiRouter.UserRouter.BASE_URL)
public class UserController {

    private final UserService userService;

    @Autowired
    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping(ApiRouter.UserRouter.FIND_BY_ID)
    public ResponseEntity<UserDTO> getById(@PathVariable Long id) {
        UserDTO userDTO = userService.findById(id);
        return ResponseEntity.ok(userDTO);
    }

    @PostMapping
    public ResponseEntity<UserDTO> create(@RequestBody Optional<UserEntity> userEntity) {
        userEntity.get().setRole(Role.USER);
        UserDTO userDTO = userService.save(userEntity);
        return ResponseEntity.ok(userDTO);
    }

    @PutMapping(ApiRouter.UserRouter.UPDATE)
    public ResponseEntity<UserDTO> update(@PathVariable("id") Optional<UserEntity> userEntityFromDb,
                           @RequestBody Optional<UserEntity> userEntity) {
        UserDTO userDTO = userService.update(userEntityFromDb, userEntity);
        return ResponseEntity.ok(userDTO);
    }

    @DeleteMapping(ApiRouter.UserRouter.DELETE_BY_ID)
    public void delete(@PathVariable("id") Optional<UserEntity> userEntity) {
        userService.delete(userEntity);
    }
}

File: src/main/java/by/hghgrtut/shop/controllers/UserController.kt
package by.hghgrtut.shop.controllers

import by.hghgrtut.shop.contracts.ApiRouter
import by.hghgrtut.shop.model.UserModel
import by.hghgrtut.shop.entities.UserEntity
import by.hghgrtut.shop.security.Role
import by.hghgrtut.shop.services.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(ApiRouter.UserRouter.BASE_URL)
class UserController @Autowired constructor(private val userService: UserService) {

    @GetMapping(ApiRouter.UserRouter.FIND_BY_ID)
    fun getById(@PathVariable id: Long?): ResponseEntity<UserModel> = getEntity(userService.findById(id))

    @PostMapping
    fun create(@RequestBody userEntity: UserEntity?): ResponseEntity<UserModel> = getEntity(userService.save(userEntity))

    @PutMapping(ApiRouter.UserRouter.UPDATE)
    fun update(@PathVariable("id") userEntityFromDb: UserEntity?, @RequestBody userEntity: UserEntity?) = getEntity(
            userService.update(userEntityFromDb, userEntity)
    )

    @DeleteMapping(ApiRouter.UserRouter.DELETE_BY_ID)
    fun delete(@PathVariable("id") userEntity: UserEntity?>?) { userService.delete(userEntity) }

    private fun getEntity(userModel: UserModel) = ResponseEntity.ok(userModel)
}